geode-core/src/main/java/org/apache/geode/internal/tcp/Connection.java

```diff
@@ -1798,6 +1794,46 @@ private void createIoFilter(SocketChannel channel, boolean clientSocket) throws
     }
   }
 
+  private SSLEngine createSslEngine(final InetSocketAddress remoteAddress) {
+    final String hostName;
+    final boolean isSender = remoteMember != null;
```

Comment:
I am not expert in this area of the code...Earlier the caller used to pass true or pass based on sender or receiver connection; is the assumption is right; the earlier argument "clientSocket" passed reflects (or based on) remoteMember setting?

Reply:
`createIoFilter` is used by both sender and receiver and that creates some confusion: it isn’t obvious what `clientSocket` really means, who and why sets it to true, and in what cases `remoteMember` can be null.
Yes, before this change, the sender set `clientSocket` to true and the receiver set it to false. The sender also sets the `remoteMember` while the receiver doesn’t (it only knows the remote address from the socket channel). So there is a correlation between `remoteMember` and `clientSocket`, and we wanted to “document” it in the code.

Reply:
Can this be a possibility: by the time createSslEngine is called the remote member has left the cluster (or forced-disconnected)? or SocketChannel is closed...
Is it a good idea to have a test, where the remote member is made to exit; right before createSslEngine is called.
Sorry if I am suggesting cases thats not practical...Please ignore if thats the case...